### PR TITLE
Fixes #2694

### PR DESCRIPTION
**Pull request by Shade-**
#2694

### CODE REVIEW

File: inc/functions_online.php [hunk 1]
```diff
@@ -161,10 +161,26 @@ function fetch_wol_activity($location, $nopermission=false)
...
+					if(!empty($user['uid']))
+					{
+						$parameters['uid'] = $user['uid'];
+					}
+					elseif(!empty($memprofile['uid']))
+					{
+						$parameters['uid'] = $memprofile['uid'];
+					}
+				}
+				elseif(!isset($parameters['uid']))
```
Comment (Devilshakerz) [inline inc/functions_online.php]: This condition should be executed independently, before `if($parameters['uid'] == 0)`

File: inc/functions_online.php [hunk 2]
```diff
@@ -161,10 +161,26 @@ function fetch_wol_activity($location, $nopermission=false)
...
 				{
 					$parameters['uid'] = 0;
 				}
+
 				$parameters['uid'] = (int)$parameters['uid'];
```
Comment (Devilshakerz) [inline inc/functions_online.php]: The casting should be done before checking `if($parameters['uid'] == 0)`, e.g.:
```php
if(isset($parameters['uid']))
{
	$parameters['uid'] = (int)$parameters['uid'];
}
else
{
	$parameters['uid'] = 0;
}

if($parameters['uid'] == 0)
{
```

File: inc/functions_online.php (hunk 1 above)
Comment (Shade-) [inline inc/functions_online.php]: Thank you @Devilshakerz, updated

File: inc/functions_online.php [hunk 3]
```diff
@@ -176,13 +182,7 @@ function fetch_wol_activity($location, $nopermission=false)
...
 					}
 				}
-				elseif(!isset($parameters['uid']))
-				{
-					$parameters['uid'] = 0;
-				}
-
-				$parameters['uid'] = (int)$parameters['uid'];
-				if($parameters['uid'] > 0)
+				elseif($parameters['uid'] > 0)
```
Comment (Devilshakerz) [inline inc/functions_online.php]: this should be just `if` as `$uid_list` is not filled correctly now

Comment (Shade-) [inline inc/functions_online.php]: Holy moly. Did a blind fix without testing it. Thank you again mate! Fixed.

**Pull request merged**